tests/test_all_rules.py

```diff
@@ -39,7 +39,7 @@ def test_file_names(self):
         self.assertIsNone(re.match(file_pattern, 'still_not_a_valid_file_name.not_json'),
                           f'Incorrect pattern for verifying rule names: {file_pattern}')
 
-        for rule in self.all_rules:
+        for rule in self.production_rules:
```

Comment:
This makes the tests more accurate but most test still fail since the `RuleCollection.default()` attempts to load and validate deprecated rules. We discussed excluding this validation in the past @rw-access but I believe deferred until necessary. There are several places to approach bypassing, so we should agree where it makes most sense

Reply:
Does this mean development rules are now completely exempt from testing? Hmm..

Does it make more sense to detect deprecation in the loader and then deserialize those more loosely? Like deserialize as a dict then do a quick check for deprecation status?

Reply:
We can make the default loader ignore deprecation as well?